src/fights.ts

```diff
@@ -738,6 +739,54 @@ const freeFightSources = [
     }
   ),
 
+  new FreeFight(
+    () => (wantPills() ? clamp(5 - get("_saberForceUses"), 0, 3 - get("_monstersMapped")) : 0),
+    () => {
+      ensureEffect($effect`Transpondent`);
+      if (have($familiar`Red-Nosed Snapper`)) cliExecute(`snapper ${$phylum`dude`}`);
+      setChoice(1387, 3);
+      if (
+        have($skill`Comprehensive Cartography`) &&
+        get("_monstersMapped") <
+          (getBestFireExtinguisherZone() && get("_fireExtinguisherCharge") >= 10 ? 2 : 3)
```

Comment:
Why does fire extinguisher charge count matter?

Reply:
We want to be able to map to use polar vortex later; i should perhaps add this in with comments, because it is entirely non-obvious